vllm/spec_decode/spec_decode_worker.py

```diff
@@ -0,0 +1,347 @@
+from typing import List, Tuple, Optional, Dict
```

Comment:
Currently, there is some complexity by separating the ways we can spec and non_spec sequences. In the future, we will remove the complexity by introducing variable proposed length and flashInfer kernel. Maybe we can add some comments about this?

Reply:
Good point, will add some comment!